### ISSUE TITLE
Drop channel-pair ops and types from library

### ISSUE THREAD

**Issue by astewart-xmos** [xmos/lib_xs3_math#58]
**TLDR**: channel pairs types and operators will no longer be supported by this library.

After some time spent considering issue #49 and after a couple conversations with people, I've decided that the channel-pair concept is too narrowly tailored for a single purpose to be a good fit for this library.

Specifically, that purpose is the stereo FFT, which allows an FFT to be applied to two real vectors significantly faster than just applying the mono FFT to each vector.  This optimization requires that the samples from the two vectors be interleaved into a single vector: hence, the type `ch_pair_s32_t`.

First, it became clear that the BFP type `bfp_ch_pair_s32_t` would need to track separate exponents (and likely headroom) for the two channels. Then, we found that were were going to need a handful of additional operators (such as power computation) that would need to be applied in a per-channel basis.  These operators would be inefficient (compared to the `bfp_s32_t` versions) and a lot of extra work to develop and maintain -- all to allow for a speed-up in the FFT in high-level user code.

Instead, we've decided that for code which uses the high-level BFP API, real time-domain vectors should just use `bfp_s32_t`, and if the stereo FFT is desired, the elements could be interleaved as the stereo FFT is called.  If this cost is unacceptable for a particular application, users are free to interleave their samples and directly make use of the low-level API to apply the stereo FFT.

This will make the library cleaner and less potentially confusing for users, and will make development faster and easier to maintain.
